pkg/sysregistriesv2/shortnames.go

```diff
@@ -0,0 +1,293 @@
+
+}
+
+// parseAliase parses the specified alias into a reference.Named.  The alias is
+// expected to not be tagged or carry a digest and *must* include a
+// domain/registry.
+func parseAlias(alias string) (reference.Named, error) {
+	repo, err := reference.Parse(alias)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot parse alias: %q", alias)
+	}
+
+	if _, ok := repo.(reference.Digested); ok {
+		return nil, errors.Errorf("invalid alias %q: must not contain digest", alias)
+	}
+
+	if _, ok := repo.(reference.Tagged); ok {
+		return nil, errors.Errorf("invalid alias %q: must not contain tag", alias)
+	}
+
+	named, ok := repo.(reference.Named)
+	if !ok {
+		return nil, errors.Errorf("invalid alias %q: must contain registry and repository", alias)
+	}
+
+	registry := reference.Domain(named)
+	if !(strings.ContainsAny(registry, ".:") || registry == "localhost") {
+		return nil, errors.Errorf("invalid alias %q: must contain registry and repository", alias)
+	}
+	return named, nil
+}
+
+// validateShortName parses the specified `name` of an alias (i.e., the left-hand
+// side) and checks if it's a short name and does not include a tag or digest.
+func validateShortName(name string) error {
+	repo, err := reference.Parse(name)
+	if err != nil {
+		return errors.Wrapf(err, "cannot parse short name: %q", name)
+	}
+
+	if _, ok := repo.(reference.Digested); ok {
+		return errors.Errorf("invalid short name %q: must not contain digest", name)
+	}
+
+	if _, ok := repo.(reference.Tagged); ok {
+		return errors.Errorf("invalid short name %q: must not contain tag", name)
+	}
+
+	named, ok := repo.(reference.Named)
+	if !ok {
+		return nil
+	}
+
+	registry := reference.Domain(named)
+	if strings.ContainsAny(registry, ".:") || registry == "localhost" {
+		return errors.Errorf("invalid short name %q: must not contain registry", name)
+	}
+	return nil
+}
+
+// parseAndValidate parses and validates all entries in conf.Aliases and stores
+// the results in conf.namedAliases.
+func (conf *shortNameAliasConf) parseAndValidate() error {
+	if conf.namedAliases == nil {
+		conf.namedAliases = make(map[string]reference.Named)
+	}
+	errs := []error{}
+	for name, alias := range conf.Aliases {
+		if err := validateShortName(name); err != nil {
+			errs = append(errs, err)
+		}
+
+		named, err := parseAlias(alias)
+		if err != nil {
+			// We want to report *all* malformed entries to avoid a
+			// whack-a-mole for the user.
+			errs = append(errs, err)
+		}
+		if len(errs) == 0 {
+			// Don't bother if there's already an error.
+			conf.namedAliases[name] = named
+		}
+	}
+	var err error // nil if no errors
+	for _, e := range errs {
+		if err == nil {
+			err = e
+		} else {
+			err = errors.Wrapf(err, "%v\n", e)
+		}
+	}
+	return err
+}
+
+func loadShortNameAliasConf(confPath string) (*shortNameAliasConf, error) {
+	conf := shortNameAliasConf{}
+
+	_, err := toml.DecodeFile(confPath, &conf)
+	switch err {
+	case nil:
+		if conf.Aliases == nil {
```

Comment:
Doesn’t this need to happen in the `IsNotExist` case as well?

Why the `switch` and not `if`?

Reply:
I think that's not needed any more. I'll check and simplify.

Thanks for checking!